src/test_fixture/ipa.rs

```diff
@@ -191,3 +195,80 @@ pub async fn test_ipa<F>(
         .collect::<Vec<_>>();
     assert_eq!(result, expected_results);
 }
+
+/// # Panics
+/// If any of the IPA protocol modules panic
+#[cfg(feature = "in-memory-infra")]
+pub async fn test_oprf_ipa<F>(
+    world: &super::TestWorld,
+    mut records: Vec<TestRawDataRecord>,
+    expected_results: &[u32],
+    config: IpaQueryConfig,
+) where
+    F: PrimeField + ExtendableField + IntoShares<semi_honest::AdditiveShare<F>>,
+    rand::distributions::Standard: rand::distributions::Distribution<F>,
+    semi_honest::AdditiveShare<F>: Serializable,
+{
+    use crate::{
+        ff::{Field, Gf2},
+        protocol::{
+            basics::ShareKnownValue,
+            prf_sharding::{attribution_and_capping_and_aggregation, PrfShardedIpaInputRow},
+        },
+        report::EventType,
+        secret_sharing::SharedValue,
+        test_fixture::Runner,
+    };
+
+    let user_cap: i32 = config.per_user_credit_cap.try_into().unwrap();
+    assert!(
+        user_cap & (user_cap - 1) == 0,
+        "This code only works for a user cap which is a power of 2"
+    );
+
+    //TODO(richaj) This manual sorting will be removed once we have the PRF sharding in place
+    records.sort_by(|a, b| b.user_id.cmp(&a.user_id));
+
+    let result: Vec<F> = world
+        .semi_honest(
+            records.into_iter(),
+            |ctx, input_rows: Vec<OprfReport<Timestamp, BreakdownKey, TriggerValue>>| async move {
+                let sharded_input = input_rows
+                    .into_iter()
+                    .map(|single_row| {
+                        let is_trigger_bit_share = if single_row.event_type == EventType::Trigger {
```

Comment:
Consider adding a comment here to explain this is temporary and will go away once we have a real
1. Decrypt
2. Add dummy records
3. Shuffle
4. Compute OPRF